src/associations/has-many.ts

```diff
@@ -0,0 +1,834 @@
+
+    if (targetInstances.length === 0) {
+      return;
+    }
+
+    const update = {
+      [this.foreignKey]: sourceInstance.get(this.sourceKey),
+      ...this.scope,
+    } as UpdateValues<T>;
+
+    const where = {
+      [this.target.primaryKeyAttribute]: targetInstances.map(unassociatedObject => {
+        return unassociatedObject.get(this.target.primaryKeyAttribute);
+      }),
+    };
+
+    await this.target.unscoped().update(update, { ...options, where });
+  }
+
+  /**
+   * Un-associate one or several target rows.
+   *
+   * @param sourceInstance instance to un associate instances with
+   * @param targetInstances Can be an Instance or its primary key, or a mixed array of instances and primary keys
+   * @param options Options passed to `target.update`
+   */
+  async remove(
+    sourceInstance: S,
+    targetInstances: AllowArray<T | Exclude<T[TargetPrimaryKey], any[]>>,
+    options: HasManyRemoveAssociationsMixinOptions<T> = {},
+  ): Promise<void> {
+    if (targetInstances == null) {
+      return;
+    }
+
+    if (!Array.isArray(targetInstances)) {
+      targetInstances = [targetInstances];
+    }
+
+    if (targetInstances.length === 0) {
+      return;
+    }
+
+    // TODO: if foreign key cannot be null, delete instead (maybe behind flag) - https://github.com/sequelize/sequelize/issues/14048
+    const update = {
+      [this.foreignKey]: null,
+    } as UpdateValues<T>;
+
+    const where = {
+      [this.foreignKey]: sourceInstance.get(this.sourceKey),
+      [this.target.primaryKeyAttribute]: targetInstances.map(targetInstance => {
+        if (targetInstance instanceof this.target) {
+          return (targetInstance as T).get(this.target.primaryKeyAttribute);
+        }
+
+        // raw entity
+        if (isPlainObject(targetInstance) && this.target.primaryKeyAttribute in targetInstance) {
+          // @ts-expect-error
+          return targetInstance[this.target.primaryKeyAttribute];
+        }
+
+        // primary key
+        return targetInstance;
+      }),
+    };
+
+    await this.target.unscoped().update(update, { ...options, where });
+  }
+
+  /**
+   * Create a new instance of the associated model and associate it with this.
+   *
+   * @param sourceInstance source instance
+   * @param values values for target model instance
+   * @param options Options passed to `target.create`
+   */
+  async create(
+    sourceInstance: S,
+    // @ts-expect-error -- {} is not always assignable to 'values', but Target.create will enforce this, not us.
+    values: CreationAttributes<T> = {},
+    options:
+      | HasManyCreateAssociationMixinOptions<T>
+      | HasManyCreateAssociationMixinOptions<T>['fields'] = {},
+  ): Promise<T> {
+    if (Array.isArray(options)) {
+      options = {
+        fields: options,
+      };
+    }
+
+    if (this.scope) {
+      for (const attribute of Object.keys(this.scope)) {
+        // TODO: fix the typing of {@link AssociationScope}
```

Comment:
When do you think we should take care of this? Before or after the release of v7?

Reply:
Either is fine. It wouldn't be a breaking change if done properly